drivers/ad9208/ad9208.c

```diff
@@ -0,0 +1,514 @@
+ *    on or directly connected to an Analog Devices Inc. component.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+/******************************************************************************/
+/***************************** Include Files **********************************/
+/******************************************************************************/
+
+#include <errno.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include "ad9208.h"
+#include <inttypes.h>
+#include "api_def.h"
+
+/**
+ * Spi write and read compatible with ad9208 API
+ * @param user_data
+ * @param wbuf Pointer to array with the data to be sent on the SPI
+ * @param rbuf Pointer to array where the data to which the SPI will be written
+ * @param len The size in bytes allocated for each of the indata and outdata arrays.
+ * @return 0 for success, any non-zero value indicates an error
+ */
+static int ad9208_spi_xfer(void *user_data, uint8_t *wbuf,
+			   uint8_t *rbuf, int len)
+{
+	struct spi_desc *spi = user_data;
+	uint8_t * buffer = (uint8_t *) malloc(len);
+	int32_t ret;
+
+	buffer[0] = wbuf[0];
+	buffer[1] = wbuf[1];
+	buffer[2] = wbuf[2];
+
+	ret = spi_write_and_read(spi, buffer, len);
+	if (ret < 0)
+		printf("Read Error %"PRId32, ret);
+	else
+		memcpy(rbuf, buffer, len);
+
+	free(buffer);
+
+ 	return ret;
+}
+
+/**
+ * Set the test mode compatible with ad9208 API
+ * @param st - The device structure
+ * @param channel - Channel number
+ * @param mode - Test mode. Accepted values:
+ * 		 AD9208_TESTMODE_OFF
+ * 		 AD9208_TESTMODE_MIDSCALE_SHORT
+ * 		 AD9208_TESTMODE_POS_FULLSCALE
+ * 		 AD9208_TESTMODE_NEG_FULLSCALE
+ * 		 AD9208_TESTMODE_ALT_CHECKERBOARD
+ * 		 AD9208_TESTMODE_PN23_SEQ
+ * 		 AD9208_TESTMODE_PN9_SEQ
+ * 		 AD9208_TESTMODE_ONE_ZERO_TOGGLE
+ * 		 AD9208_TESTMODE_USER
+ * 		 AD9208_TESTMODE_RAMP
+ * @return 0 for success, any non-zero value indicates an error
+ */
+static int ad9208_testmode_set(struct ad9208_state *st,
+			       unsigned int chan,
+			       unsigned int mode)
+{
+	ad9208_handle_t *ad9208_h = st->adc_h;
+	int ret;
+
+ 	ret = ad9208_adc_set_channel_select(ad9208_h, BIT(chan & 1));
```

Comment:
does this work only for channels 1, 3, 5 ? 
BIT(chan & 1)

Reply:
I think it's implemented like that because there are only 2 channels that you should be able to set: channel 0 and channel 1